### DISCUSSION TITLE
Does this also include the ability to filter specific record types?

### DISCUSSION THREAD

**Discussion by eabhijith** [jkranz-rk/RecordTypePicker#29] [Q&A] [answered]
Hey,

Does this package also include the ability to filter specific record types?

**Comment by jkranz-rk** [comment, marked as answer]
At the moment, no, the only filtering it does is filter by user access, but that is something I'd love to implement.

**Comment by jkranz-rk** [comment]
@eabhijith Could you elaborate a bit on your use case? I have this on my roadmap, but would like to validate that against how you'd like to use it. What I had been thinking was that you'd be able to define (at build-time) specific "allowed" record types, thereby filtering out any others (along with applying user permissions for the record types). Would something like that fit your use case?

**Reply by eabhijith** [reply]
Hey @jkranz-rk , Thanks for the response mate, and happy new year!

Sure, My use case is to show record types filtering them on the flow for selection based on the selection (from the filtered list), then use lightning navy mix in lwc connected callback to navigate the user to the standard record creation page.

Why Filter Record types:
1. We have a selection of lookups first
2. Based on the location selection, we will have to filter record types.
3. Navigate the user to the selected record creation.

BR,
Abhijith

**Reply by eabhijith** [reply]
P:S We have implemented this using the unofficial sf lookup filter currently (UI is not great but it works for now)!

**Reply by jkranz-rk** [reply]
Hey @eabhijith, thanks for your patience on this! I'm starting work on this soon, but before I build, a couple questions for you regarding your use case:

 - How do you use the user's "location" selection to filter the available RecordTypes? Would it be sufficient for the RecordTypePicker to accept a collection as the "base" collection (rather than use all of them to evaluate User Access)? 
   - For instance, you could perform a Get on RecordType with the filtering you need to perform, then supply the RecordTypePicker with the result of the Get.
 - Would you also need an option for RecordTypePicker to bypass the user-access filtering or would you need it to continue to apply User Access permissions even with the pre-filtering?

Thanks again!

**Reply by eabhijith** [reply]
Hey @jkranz-rk , 

Thanks for writing!

1. First one is right  "you could perform a Get on RecordType with the filtering you need to perform, then supply the RecordTypePicker with the result of the Get." --> This works for our use case
2. User access bypass was a must in our case as We have apex based conditional sharing which would later decide if the user should be able to view the record or not, but every user should be able to create a request for every record type.

BR,
Abhijith

**Reply by Anthony-Waring** [reply]
Hey @jkranz-rk 

I am just reading through this discussion and was wondering how your getting the record type picker to show only those from the Get element? The component only lets me choose objects with no option for resources/variables? 

Many thanks,
Anthony
